### PR TITLE
292 add functionality to support categories in the maps dropdown

### PR DESCRIPTION
**Pull request by ifirmawan**
## TODO/DONE

- [x] Fix the loading indicator in the map after implementing the new JMP

### The maps dropdown in all instances : 
- [x] Uganda
- [x] Ethiopia
- [x] Nepal
- [x] Bangladesh

### CODE REVIEW

File: backend/source/wai-bangladesh/config.js
Comment (dedenbangkit) [inline backend/source/wai-bangladesh/config.js:1]: Can this still support question id too?

File: frontend/src/pages/main/MainMaps.jsx [hunk 1]
```diff
@@ -482,28 +430,63 @@ const MainMaps = ({ question, current }) => {
...
+              const markOptions = findMarker?.option || findMarker?.labels;
+              return {
+                ...findMarker,
+                ...md,
+                option: markOptions,
+              };
+            });
+            setMarkerOptions(_markerOptions);
+
+            if (typeof selectableMarkerQuestion?.id === 'undefined') {
```
Comment (dedenbangkit) [inline frontend/src/pages/main/MainMaps.jsx:452]: Could you please explain why we need to check the type of `selectableMarkerQuestion?.id` ?

**Review by dedenbangkit — COMMENTED** [review]
Please answer the individual reviews first

File: backend/source/wai-bangladesh/config.js
Comment (ifirmawan) [reply backend/source/wai-bangladesh/config.js:1]: Hi mas @dedenbangkit , for shape: yes, marker: no. I think I need to make some adjustments to the marker id.

File: frontend/src/pages/main/MainMaps.jsx (hunk 1 above)
Comment (ifirmawan) [reply frontend/src/pages/main/MainMaps.jsx]: after I looked again it turned out that this did not need to be checked

File: backend/source/wai-bangladesh/config.js
Comment (ifirmawan) [reply backend/source/wai-bangladesh/config.js:1]: now, shapes and markers are supported to use question id.

**Review by dedenbangkit — APPROVED** [review]

**Pull request merged**